[Cute, SM90] Varlen Cute implementation doesn't work on H100

Running `pytest tests/cute/test_flash_attn_varlen.py` with an H100 GPU throws the following error:
```
E           cutlass.base_dsl.common.DSLRuntimeError: DSLRuntimeError: 🧊🧊🧊 ICE IR Verification Failed 🧊🧊🧊
E             Caused exception: Verification failed:
E           error: "cute.copy("("/opt/x/lib/python3.12/site-packages/flash_attn/cute/flash_fwd.py":456:20): 'cute.copy' op '!cute_nvgpu.atom.universal_copy<f16, 128 b>' ptr alignment does not meet requirement
E            note: "cute.copy("("/opt/x/lib/python3.12/site-packages/flash_attn/cute/flash_fwd.py":456:20): see current operation: "cute.copy"(%arg18, %2461, %2473) : (!cute.tiled_copy<!cute_nvgpu.atom.universal_copy<f16, 128 b>, layout_copy_tv = <"((8,32),8):((256,1),32)">, tiler_mn = <"[32:1;64:1]">>, !cute.memref<f16, rmem, align<32>, "((8,1),(1)):((1,0),(0))">, !cute.memref<f16, gmem, "((8,1),(1)):((1,0),(0))">) -> ()
```
From what I gathered the problem is [this line](https://github.com/Dao-AILab/flash-attention/blob/a0f9f418fd15df20d9418daca715ede24387c412/flash_attn/cute/flash_fwd.py#L398-L399)

```python
mO_cur = cute.domain_offset((offset, 0), mO[None, None, head_idx])
```
`mO` when printed is `tensor<ptr<f16, gmem, align<16>> o (?,?,?):(?,1,?)>`
`mO_cur` when printed is `tensor<ptr<f16, gmem> o (?,?):(?,1)>`

I tried adding `cute.assume` for `offset`, but it doesn't help.

--- a/flash_attn/cute/flash_bwd.py
+++ b/flash_attn/cute/flash_bwd.py
@@ -385,7 +385,7 @@ class FlashAttentionBackwardSm80:
                            for t in (mQ, mK, mV, mdO, mLSE, mdPsum, mdQaccum, mdK, mdV, mCuSeqlensQ, mCuSeqlensK, mSeqUsedQ, mSeqUsedK)))
         # Assume all strides are divisible by 128 bits except the last stride
         # Skip cute.assume() for stride=0 (broadcast dims from expand() are Python ints)
-        new_stride = lambda t: (*(cute.assume(s, divby=128 // t.element_type.width) if s != 0 else s for s in t.stride[:-1]), t.stride[-1])
+        new_stride = lambda t: (*(cute.assume(s, divby=128 // t.element_type.width) if not isinstance(s, int) or s != 0 else s for s in t.stride[:-1]), t.stride[-1])
         mQ, mK, mV, mdO, mLSE, mdPsum, mdQaccum, mdK, mdV = [cute.make_tensor(t.iterator, cute.make_layout(t.shape, stride=new_stride(t))) if t is not None else None for t in (mQ, mK, mV, mdO, mLSE, mdPsum, mdQaccum, mdK, mdV)]
         self.varlen_q = (mCuSeqlensQ is not None)
         self._setup_attributes()
@@ -401,7 +401,7 @@ class FlashAttentionBackwardSm80:
             TileScheduler = SingleTileScheduler
             num_batch = mK.shape[0]
 
-        # Uses seqlen k, etc. since main bwd kernel's blocks are over n 
+        # Uses seqlen k, etc. since main bwd kernel's blocks are over n
         tile_sched_args = TileSchedulerArguments(
             num_block=cute.ceil_div(mK.shape[1], self.n_block_size),
             num_head=num_head,
@@ -416,7 +416,7 @@ class FlashAttentionBackwardSm80:
             mCuSeqlensQ=mCuSeqlensK,
             mSeqUsedQ=mSeqUsedK,
         )
-        
+
         tile_sched_params = TileScheduler.to_underlying_arguments(tile_sched_args)
         grid_dim = TileScheduler.get_grid_shape(tile_sched_params)
 
@@ -1000,7 +1000,7 @@ class FlashAttentionBackwardSm80:
         num_head: cutlass.Int32,
         batch_size: cutlass.Int32,
         seqlen: SeqlenInfoQK,
-        d_head: cutlass.Int32, 
+        d_head: cutlass.Int32,
         d_head_v: cutlass.Int32
     ):
         rdV = cute.make_fragment_like(acc_dV, self.dtype)

--- a/flash_attn/cute/flash_bwd_sm90.py
+++ b/flash_attn/cute/flash_bwd_sm90.py
@@ -354,7 +354,9 @@ class FlashAttentionBackwardSm90:
         # Skip cute.assume() for stride=0 (broadcast dims from expand() are Python ints)
         new_stride = lambda t: (
             *(
-                cute.assume(s, divby=128 // t.element_type.width) if s != 0 else s
+                cute.assume(s, divby=128 // t.element_type.width)
+                if not isinstance(s, int) or s != 0
+                else s
                 for s in t.stride[:-1]
             ),
             t.stride[-1],

--- a/flash_attn/cute/flash_fwd.py
+++ b/flash_attn/cute/flash_fwd.py
@@ -663,7 +663,7 @@ class FlashAttentionForwardSm80(FlashAttentionForwardBase):
         new_stride = lambda t: (
             *(
                 cute.assume(s, divby=128 // t.element_type.width)
-                if s != 0
+                if not isinstance(s, int) or s != 0
                 else s
                 for s in t.stride[:-1]
             ),

